src/ShipEngine.php

```diff
@@ -70,22 +78,35 @@ public function formatAddress($address)
      * @param array $addresses Array of Address\Address or domain Addresses which are passed through the Formatter
      *
      * @return AddressVerification\VerificationResult[] Array of Verification Results for every address requested
-     * @throws Exception\ApiErrorResponse
-     * @throws Exception\ApiRequestFailed
+     * @throws RequestException
+     * @throws ClientException
      */
     public function validateAddresses(array $addresses)
     {
-        $addressData = array_map(function ($address) {
-            return $address instanceof Address\Address
-                ? $address->toArray()
-                : $this->addressFactory->factory($address)->toArray();
-        }, $addresses);
+        $addressData = $this->addressFactory->getAddressData($addresses);
 
-        $response = $this->requestFactory->validateAddresses($addressData)->send();
+        $response = $this->client->post('addresses/validate', [
+            'json' => $addressData
+        ]);
 
         return array_map(function ($data) {
             return new AddressVerification\VerificationResult($data);
-        }, $response->getData());
+        }, json_decode($response->getBody()->getContents(), 1));
```

Comment:
Refactoring this to a function to pull response data and use whatever key we want to pull from (for one layered arrays only for now).

Reply:
Alright, Changed.